ui/admin/app/routes/scopes/scope/credential-stores/credential-store/credentials/new.js

```diff
@@ -19,17 +19,39 @@ export default class ScopesScopeCredentialStoresCredentialStoreCredentialsNewRou
 
   /**
    * Creates a new unsaved credential in current credential store.
+   * Also rollback/destroy any new, unsaved instances from this route before
+   * creating another, but reuse name/description if available.
    * @return {CredentialModel}
    */
   model({ type = 'username_password' }) {
     const { id: credential_store_id } = this.modelFor(
       'scopes.scope.credential-stores.credential-store'
     );
+    let name, description;
+    if (this.currentModel?.isNew) {
+      ({ name, description } = this.currentModel);
+      this.currentModel.rollbackAttributes();
+    }
     return this.store.createRecord('credential', {
       type,
       credential_store_id,
+      name,
+      description,
     });
   }
+
+  /**
+   * Set query param to default type
+   * before exiting route
+   * @param {Controller} controller
+   * @param {boolean} isExiting
+   */
+  resetController(controller, isExiting) {
+    if (isExiting) {
+      controller.set('type', 'username_password');
+    }
+  }
```

Comment:
`queryParam` values are saved even after a user exits the route. Added the above method to reset the queryParam value so that the correct card is selected when user re-enters the route to create a new credential. @cameronperera @DhariniJeeva thoughts on this.

Reply:
Nice. I dig this approach. I didn't realize this is a problem we have but all of our routes that use a queryParam have the same issue. It even persist upon successful creation of said resource. When you go back to create a new resource, it will still be the same type of the previously created resource.

Reply:
Thank you for noticing this issue!

Reply:
Stickiness may or may not be desirable.  For example, if a user wants to create a bunch of items of the same type having to change it each time might actually be undesirable.  Before we implement this UX here, lets discuss.  If we decide to do it, it ought to be holistic (there are other places where this change would need to be made).

Reply:
Good point, I will remove this.